### PR TITLE
Ching Direct Message Step 1 to 4

### PR DESCRIPTION
**Pull request by huang27c**
I've added code for Direct message from step 1 to 4.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -52,23 +53,25 @@ public void storeMessage(Message message) {
...
    */
-  public List<Message> getMessages(String user) {
+  public List<Message> getMessages(String recipient) {
     List<Message> messages = new ArrayList<>();
 
     Query query =
-        new Query("Message")
-            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
-            .addSort("timestamp", SortDirection.DESCENDING);
+            new Query("Message")
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Datastore.java]: It looks like you increased the level of indentation when editing.  You should probably undo that.

File: src/main/java/com/google/codeu/data/Message.java [hunk 2]
```diff
@@ -30,15 +32,18 @@
...
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  //add recipient to the constructor
+  public Message(String user, String text, String recipient) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
   }
 
-  public Message(UUID id, String user, String text, long timestamp) {
+  //add recipient to the constructor
+  public Message(UUID id, String user, String text, long timestamp, String recipient) {
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Javadoc comments are in the style:

/**
/* Constructs a new message.
/* 
/* @param id A non-null uuid to uniquely identify the constructed message.
...

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 3]
```diff
@@ -77,10 +77,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    //get the value of query parameter
```
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: This comment doesn't add much to the following line.

**Review by nnunley — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: I've added comment for every method.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 3 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: I removed this comment.

File: src/main/java/com/google/codeu/data/Message.java [hunk 4]
```diff
@@ -29,15 +29,21 @@
   private long timestamp;
 
   /**
-   * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
+   * Constructs a new {@link Message} posted by {@code user} with {@code text} content to {@code recipient}. Generates a
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: I think if you run this through the google-code-formatter code that I linked to, that will fix the wrapping issue that you're seeing.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 5]
```diff
@@ -52,23 +53,25 @@ public void storeMessage(Message message) {
...
+                    .setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient))
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Datastore.java]: Looks like you need to reformat this file.

File: src/main/java/com/google/codeu/data/Message.java [hunk 6]
```diff
@@ -30,15 +32,18 @@
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
    * random ID and uses the current system time for the creation time.
    */
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  //add recipient to the constructor
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Delete this line

File: src/main/java/com/google/codeu/data/Message.java [hunk 7]
```diff
@@ -30,15 +32,18 @@
...
+  public Message(String user, String text, String recipient) {
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Update the constructor comment to include @param javadocs.

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Looks like you win here, and have to add new javadoc for this constructor.

Intellij should actually do the right thing if you type /** <tab>  in terms of adding the @param values of each of the arguments.  You'll still have to add high level documentation describing what this constructor is supposed to do.

File: src/main/java/com/google/codeu/data/Message.java [hunk 8]
```diff
@@ -56,4 +61,7 @@ public String getText() {
   public long getTimestamp() {
     return timestamp;
   }
+
+  //add getter for recipient that provides access to the field
+  public String getRecipient(){ return recipient; }
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Looks like another thing that should be caught by reformatting the file this is in.

**Review by nnunley — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/data/Message.java [hunk 9]
```diff
@@ -52,36 +53,28 @@ public Message(UUID id, String user, String text, long timestamp, String recipie
     this.recipient = recipient;
   }
 
-  /**
-   * @return current non-null UUID
-   */
+  /** @return current non-null UUID */
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Add a summary for this.  "Returns a current non-null UUID." is a valid fix.

File: src/main/java/com/google/codeu/data/Message.java [hunk 10]
```diff
@@ -52,36 +53,28 @@ public Message(UUID id, String user, String text, long timestamp, String recipie
...
   public UUID getId() {
     return id;
   }
 
-  /**
-   * @return the sender of the user
-   */
+  /** @return the sender of the user */
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: Same here

File: src/main/webapp/index.html [hunk 11]
```diff
@@ -21,18 +21,55 @@
...
+        color:white;
+        font-size: 40px
+      }
+      p{
+        color: white;
+      }
+      .center {
+        margin: auto;
+        width: 15%;
+      }
```
Comment (nnunley) [inline src/main/webapp/index.html]: Pull this in to the css/main.css file.

Comment (nnunley) [reply src/main/webapp/index.html]: Additionally, this should probably be in a different PR/commit

File: src/main/webapp/index.html [hunk 12]
```diff
@@ -21,18 +21,55 @@
...
+    </style>
   </head>
   <body onload="addLoginOrLogoutLinkToNavigation();">
     <nav>
       <ul id="navigation">
         <li><a href="/">Home</a></li>
       </ul>
     </nav>
-    <h1>CodeU Starter Project</h1>
+    <hr style="width: 1440px">
```
Comment (nnunley) [inline src/main/webapp/index.html]: You probably want to add this to your css as well.

File: src/main/webapp/index.html [hunk 13]
```diff
@@ -21,18 +21,55 @@
...
+    <div style="display: flex; justify-content: center;"><img src="codeU.png" alt="CodeU logo" class="center"></div>
```
Comment (nnunley) [inline src/main/webapp/index.html]: Hrm.  You probably want to add a new css rule in your header

File: src/main/webapp/user-page.html [hunk 14]
```diff
@@ -22,15 +22,42 @@
...
+      div{
+        color:white;
+      }
+      #page-title{
+        color: white;
+      }
+      #message-form{
+        color: white;
+      }
+
```
Comment (nnunley) [inline src/main/webapp/user-page.html]: Add this to your user-page.css

File: src/main/webapp/user-page.html [hunk 15]
```diff
@@ -22,15 +22,42 @@
...
+    </Style>
   </head>
   <body onload="buildUI();">
     <nav>
       <ul id="navigation">
         <li><a href="/">Home</a></li>
       </ul>
     </nav>
+    <hr style="width: 1440px">
```
Comment (nnunley) [inline src/main/webapp/user-page.html]: switch this and the previous hr from a style to a class

**Review by nnunley — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/data/Message.java (hunk 6 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: Deleted.

File: src/main/java/com/google/codeu/data/Message.java (hunk 7 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: Added.

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: Got it!

File: src/main/java/com/google/codeu/data/Message.java (hunk 8 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: I believe this is fixed.

File: src/main/java/com/google/codeu/data/Message.java (hunk 9 above)
Comment (huang27c) [reply src/main/java/com/google/codeu/data/Message.java]: Do you mean change it to /**Returns a current non-null UUID*/ or 
/**Returns a current non-null UUID.
* @retunrs current non-null UUID*/

**Review by nnunley — APPROVED** [review]

File: src/main/java/com/google/codeu/data/Message.java (hunk 9 above)
Comment (nnunley) [reply src/main/java/com/google/codeu/data/Message.java]: You unfortunately need to double the text here to make the linter happy.

So yes, you'e right.

**Review by nnunley — APPROVED** [review]

**Pull request merged**